app.js

```diff
@@ -60,10 +60,24 @@ _.each(routes, (verbs, url) => {
         if (!req.authUser) {
           next(new errors.HttpStatusError(401, 'Authentication required!'))
         }
-        // Access check
-        if (Array.isArray(def.access) &&
-          (!req.authUser.roles || !_.intersection(req.authUser.roles, def.access).length)) {
-          next(new errors.HttpStatusError(403, 'You are not allowed to perform this action!'))
+
+        if (req.authUser.roles) {
+          if (Array.isArray(def.access) &&
+            !_.intersection(req.authUser.roles, def.access).length) {
+            next(new errors.HttpStatusError(403, 'You are not allowed to perform this action!'))
+          } else {
+            next()
+          }
+        } else if (req.authUser.scopes) {
+          if (Array.isArray(def.scopes) &&
+            !def.scopes.some(s => req.authUser.scopes.indexOf(s) !== -1)) {
```

Comment:
are we doing an in string check here instead of splitting the scopes by spaces and checking it as an array?
I could foresee this being an issue if we created scopes in the future like `read:submission` and `read:submission_self`. The `indexOf()` would equate the former with the latter which would be bad. I think this should be exact match, normalized to lowercase, only.